Assets/Scripts/ResourceSystem/ResourceBuildingsController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Growing.ResourceSystem
+{
+    public sealed class ResourceBuildingsController : IDisposable
+    {
+        private readonly ResourceBuildingsHolder resourceBuildingsHolder;
+        private readonly ResourceHolder resourceHolder;
+
+        private readonly CancellationTokenSource cts = new();
+
+        public ResourceBuildingsController(ResourceBuildingsHolder resourceBuildingsHolder,
+            ResourceHolder resourceHolder)
+        {
+            this.resourceBuildingsHolder = resourceBuildingsHolder;
+            this.resourceHolder = resourceHolder;
+
+            RunCreationCycle();
+        }
+
+        private void RunCreationCycle()
+        {
+            Task.Run(ResourceCreationCycle);
+        }
+
+        private async Task ResourceCreationCycle()
+        {
+            while (!cts.IsCancellationRequested)
+            {
+                CreateResources();
+                await Task.Delay(1000);
```

Comment:
The Unity SynchronizationContext will proceed awaited Task execution in the main thread

Reply:
Done